### PR TITLE
MISUV-7735: Claim to Adjust POA | Build - Success message page after POAs are updated

### PR DESCRIPTION
**Pull request by omirostr1**

### CODE REVIEW

File: app/controllers/claimToAdjustPoa/SuccessController.scala [hunk 1]
```diff
@@ -16,29 +16,60 @@
...
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.{AuthenticatorPredicate, ClaimToAdjustUtils}
+import views.html.claimToAdjustPoa.SuccessView
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class SuccessController @Inject()(val authorisedFunctions: AuthorisedFunctions,
-                                  val auth: AuthenticatorPredicate,
+                                  val view: SuccessView,
```
Comment (avelx) [inline app/controllers/claimToAdjustPoa/SuccessController.scala:35]: I think we need to rename this controller to something else, so to easy up reading this in the future

Comment (omirostr1) [reply app/controllers/claimToAdjustPoa/SuccessController.scala:35]: Does "SuccessMessagePageController" sounds better?

Comment (avelx) [reply app/controllers/claimToAdjustPoa/SuccessController.scala:35]: I think any reference to ClaimToAdjust sub-domain would nice to have, as its not clear from the name what this is related to

File: app/controllers/claimToAdjustPoa/SuccessController.scala [hunk 2]
```diff
@@ -16,29 +16,60 @@
...
                                  (implicit val appConfig: FrontendAppConfig,
                                   implicit override val mcc: MessagesControllerComponents,
                                   val ec: ExecutionContext)
-  extends ClientConfirmedController {
+  extends ClientConfirmedController with ClaimToAdjustUtils{
 
-  def show(isAgent: Boolean): Action[AnyContent] =
-    auth.authenticatedAction(isAgent) {
+  private def endOfJourney(implicit hc: HeaderCarrier) = {
+    sessionService.setCompletedJourney(hc, ec)
```
[the remark this answers is not in the document]
Comment (omirostr1) [reply app/controllers/claimToAdjustPoa/SuccessController.scala:47]: Deleted the method and added the session service call inside the main show method👍🏻

File: app/views/claimToAdjustPoa/SuccessView.scala.html [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+@mainTemplate(pageTitle = getMessage("heading"),
+    isAgent = isAgent) {
+
+    @govukPanel(Panel(
+        title = Text(getMessage("h1")),
+        content = HtmlContent(getMessage("p1", poaTaxYear.startYear.toString, poaTaxYear.endYear.toString, poaTotalAmount.toCurrencyString))
+    ))
+
+    @p(id = Some("p2")) {
+        @getMessage("p2", poaTaxYear.startYear.toString, poaTaxYear.endYear.toString, (poaTaxYear.endYear + 1).toString)
```
Comment (avelx) [inline app/views/claimToAdjustPoa/SuccessView.scala.html:61]: `poaTaxYear.endYear + 1`
you can change to improve readability 
`poaTaxYear.nextYear.endYear `

File: app/controllers/claimToAdjustPoa/SuccessController.scala (hunk 1 above)
Comment (omirostr1) [reply app/controllers/claimToAdjustPoa/SuccessController.scala:35]: Okay I see, so something like SuccessMessagePoaController or SuccessForPoaJourneyController I think is more suitable

File: app/views/claimToAdjustPoa/SuccessView.scala.html (hunk 3 above)
Comment (omirostr1) [reply app/views/claimToAdjustPoa/SuccessView.scala.html:61]: Done.

File: app/services/PaymentOnAccountSessionService.scala [hunk 4]
```diff
@@ -90,4 +90,26 @@ class PaymentOnAccountSessionService @Inject()(poAAmmendmentDataRepository: PoAA
     }
   }
 
+  def setCompletedJourney(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[Throwable, Unit]] = {
+    poAAmmendmentDataRepository.get(hc.sessionId.get.value).flatMap {
+      case Some(data: PoASessionData) =>
+        val newData: PoAAmendmentData = data.poaAmendmentData match {
+          case Some(value) =>
+            value.copy(hasCompletedJourneySuccessfully = Some(true))
```
[the remark this answers is not in the document]
Comment (omirostr1) [reply app/services/PaymentOnAccountSessionService.scala:98]: Should I include the method in the 'ClaimToAdjustHelper' service?

File: app/controllers/claimToAdjustPoa/SuccessController.scala (hunk 1 above)
Comment (avelx) [reply app/controllers/claimToAdjustPoa/SuccessController.scala:35]: PaymentsOnAccountAdjustedController or PaymentsOnAccountUpdatedController, something similar would work

File: app/services/PaymentOnAccountSessionService.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (avelx) [reply app/services/PaymentOnAccountSessionService.scala:98]: ignore comment above, logic already in the required service

File: app/controllers/claimToAdjustPoa/PaymentsOnAccountAdjustedController.scala [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  def show(isAgent: Boolean): Action[AnyContent] = auth.authenticatedAction(isAgent) {
+    implicit user =>
+      ifAdjustPoaIsEnabled(isAgent) {
+        {
+          for {
+            poaMaybe <- EitherT(claimToAdjustService.getPoaForNonCrystallisedTaxYear(Nino(user.nino)))
+          } yield poaMaybe
+        }.value.flatMap {
+          case Right(Some(poa)) =>
+            sessionService.setCompletedJourney(hc, ec).flatMap {
```
[the remark this answers is not in the document]
Comment (danielmaltz1) [reply app/controllers/claimToAdjustPoa/PaymentsOnAccountAdjustedController.scala:54]: To be discussed

**Review by avelx — APPROVED** [review]
approve with comments / to be discussed

**Pull request merged**